### ISSUE TITLE
Sections without code display code - High priority

### ISSUE THREAD

**Issue by justineechen** [OpenLiberty/openliberty.io#777]
Example: While the middle panel is in the `What you'll learn` section, the right panel shows code that is not mentioned until a later section. This might be confusing for the user because there is no explanation for the code they are seeing in the middle panel. Maybe if there is a blank right panel when there is no relevant code to be displayed that would be better.

**Comment by justineechen** [comment]
The discussion thus far for this issue is that this is the design, I will be looking into a workaround later today (perhaps an empty file for sections without code files)

**Comment by gkwan-ibm** [comment]
As a user, if it is the design, I will say the design is not good.

**Comment by justineechen** [comment]
Having a blank file in place shows the include statement
<img width="1411" alt="screen shot 2018-11-13 at 2 53 15 pm" src="https://user-images.githubusercontent.com/18709242/48443181-4a0b5980-e75e-11e8-8f3a-20b57726ab30.png">

**Comment by justineechen** [comment]
Maybe we can have this kind of placeholder until we reach a file?

<img width="1680" alt="screen shot 2018-11-15 at 3 15 53 pm" src="https://user-images.githubusercontent.com/18709242/48579046-5ff95580-e8e9-11e8-82b2-2bbb903f9b53.png">

**Comment by steven1046** [comment]
@cajaygle @megmulsy what are your thoughts on displaying the github popup until one of the sections needs to show the code?

**Comment by gkwan-ibm** [comment]
@steven1046, @cajaygle, @megmulsy

Usually a guide contains multiple sections. Some sections contain/point to a file but some surely not. The main issue is that when we scroll to the section without file, the file from other section will be shown on the right pane. It is confusing. May we suggest the right pane to show nothing in this situation? I can schedule a meeting if you think it have to be discussed or want us to show you what the issue is.

Thanks

**Comment by steven1046** [comment]
I agree, we shouldn't show code not associated with the guide on the left. Waiting on design for feedback on what to show on the right.

**Comment by cajaygle** [comment]
The design intent is that the Git clone box be displayed in the righthand pane for the first sections until the end of the Getting Started section.  Currently in the Creating a RESTful Service guide, the Git clone box is only displayed at the very start of the guide -- this is not correct.  Behind the Git clone box we display the first file, but disabled and blurred.  The purpose of doing that, as opposed to displaying an empty pane, is so that first time users of a guide understand from the start that the righthand pane is going to be used for displaying code -- it orients them to the multipane design, but without showing a code file in a way that is distracting.

I agree that the way the REST guide is currently implemented, where the code file is displayed as soon as the user moves to the What You'll Learn section, is confusing.  But that implementation doesn't reflect the intended design.

As the user moves through the subsequent sections we change the righthand pane to reflect the instructions on the left.  In the interactive guides we recently made a design change that I think will be useful to bring to the static guides as well:  we changed the editor pane to be a tabbed multi-file editor.  The way that it works is this:  at the start of the guide the editor displays the first file that will be used (but disabled and blurred, as described).  When they get to the section where the file is first referenced, then the code pane is enabled.   When the user moves to a next section where a new file is used, instead of replacing the file in the code pane, we add the second file in a new editor tab and display that tab.  The first file is displayed in another tab, and the user can click between the tabs.  As the user progresses through the guide we will add new editor tabs for each new file.   New tabs are added to the left side of the tab bar, with the oldest file in the rightmost tab.  In this way the tabs become a record of the files they've worked with in the guide and they are free to switch between them.

One interaction detail:  once a tab is added it is not removed, even if the user goes back to a previous section in the guide.  For example, lets say that guide section 1 uses File 1, section 2 uses File 2, and section 3 uses File 3.  When the guide starts, the editor will have a single tab displaying File 1.  When the user moves to section 2, the editor will have two tabs:  File 2 (on left) and File 1 (on right).  At that point if the user switches back to section 1 the editor will still have File 2 and File 1, but the tab with File 1 will now be selected.  When they switch back to section 2, the tab with File 2 will be selected.  When they move to section 3, three tabs will be present, File 3, File 2, File 1.  If the user goes back to a previous section, all three tabs remain but the selected tab is changed to the file the current section is referencing.   If a section doesn't reference a particular file, then the tabs are left as-is -- no change from the previous section.

**Comment by steven1046** [comment]
#798 should fix the Github popup problem.

One thing to note about the tabbed editor on the left side of the code pane is that it may affect the number of characters per line.

**Comment by cajaygle** [comment]
<img width="788" alt="screen shot 2018-11-21 at 2 47 10 pm" src="https://user-images.githubusercontent.com/27516391/48865055-63cf2100-ed9c-11e8-86bd-21180b4c727f.png">

**Comment by cajaygle** [comment]
Tabs across the top of the editor.  Replacing the current file name line.

**Comment by gkwan-ibm** [comment]
@cajaygle  Is the tabs already implemented for the static guide or we need to wait? I think we better try it out, see whether it is acceptable or can resolve the confusing issue.

**Comment by steven1046** [comment]
The tabbed editor isn't implemented yet in the static guides.

**Comment by steven1046** [comment]
@cajaygle If the user clicks on a tab from a previous section they were on, would it scroll to that guide step or just remain on the current step so they can look at the file? They might get confused if it scrolls them to a different step but it could also confuse them if they view the file that isn't referenced in the current guide step.

**Comment by steven1046** [comment]
Is this issue handled?

**Comment by justineechen** [comment]
The issue is handled, thank you!

> @cajaygle If the user clicks on a tab from a previous section they were on, would it scroll to that guide step or just remain on the current step so they can look at the file? They might get confused if it scrolls them to a different step but it could also confuse them if they view the file that isn't referenced in the current guide step.

 @steven1046 @cajaygle If this is an issue you can open another issue for this
